### DISCUSSION TITLE
Is it possible to add additional basemaps to the map page?

### DISCUSSION THREAD

**Discussion by evanwill** [CollectionBuilder/collectionbuilder.github.io#53] [General]
Question from @writingafterthedisaster :
Currently, three basemaps are available for use: Esri_WorldStreetMap, Esri_NatGeoWorldMap, Esri_WorldImagery. Is it possible to add other basemaps?

**Comment by evanwill** [comment]
@writingafterthedisaster yes it is possible--but requires some customization of the template code. How easy that is depends on which basemaps you want to use! 

The code is in ["_includes/js/map-js.html"](https://github.com/CollectionBuilder/collectionbuilder-gh/blob/main/_includes/js/map-js.html) starting around line 35.

There is three parts to add a new one:

1. Set up a variable containing the map layer information. For some example base maps that follow the same pattern as we are using, you can copy the variable from [Leaflet Providers Preview](https://leaflet-extras.github.io/leaflet-providers/preview/).
2. add name to the base map switcher. The `baseMaps` variable sets up the names that appear in the switcher that appears in the upper right of the map. It follows the pattern of `"Display Name": Map_Layer_var_name`.
3. Set the default basemap. This is usually set in "theme.yml" as the `map-base` option. If you add a new layer variable, use that name instead of the default ones. Alternatively, in map-js.html you can edit the variable name used to load the base map (which is next after setting up the `baseMaps` var).

If you want to do more customization, check the [Leaflet docs](https://leafletjs.com/reference.html), and [Leaflet basemap providers plugins](https://leafletjs.com/plugins.html#basemap-providers).

Let me know if that helps!

**Reply by evanwill** [reply]
Also note that some of the base maps in the free [Leaflet Providers Preview](https://leaflet-extras.github.io/leaflet-providers/preview/) may have usage limitations--check the [Leaflet Providers readme for notes](https://github.com/leaflet-extras/leaflet-providers).

**Comment by writingafterthedisaster** [comment]
Very helpful. Thank you, @evanwill—I'll try this out!
